program/rust/src/rust_oracle.rs

```diff
@@ -10,10 +33,61 @@ pub fn update_price(
     instruction_data: &[u8],
     input: *mut u8,
 ) -> OracleResult {
-    //For now, we did not change the behavior of this. this is just to show the proposed structure
-    // of the program
-    c_entrypoint_wrapper(input)
+    let account_len = accounts[1].try_data_len()?;
+    let price_t_with_twap_tracker_len: usize = PRICE_ACCOUNT_SIZE
+        .try_into()
+        .map_err(|_| OracleError::IntegerCastingError)?;
+    let price_t_size = size_of::<pc_price_t>();
+    if account_len < price_t_with_twap_tracker_len {
+        if account_len != price_t_size {
+            return Err(ProgramError::InvalidArgument);
+        }
+        msg!("Please resize the account to allow for SMA tracking!");
+        return c_entrypoint_wrapper(input);
+    }
+    let c_ret_value = c_entrypoint_wrapper(input)?;
+    if c_ret_value == SUCCESSFULLY_UPDATED_AGGREGATE {
+        //update_tracker
+        msg!("updated tracker!");
+    }
+    Ok(c_ret_value)
 }
+
+/// A helper function that upgrades a price account, used by update_version
+pub fn upgrade_price_account(
+    program_id: &Pubkey,
+    accounts: &Vec<AccountInfo>,
+    instruction_data: &[u8],
+) -> ProgramResult {
+    let account_len = accounts[1].try_data_len()?;
+    let new_account_len: usize = PRICE_ACCOUNT_SIZE
+        .try_into()
+        .map_err(|_| OracleError::IntegerCastingError)?;
+    let price_t_size = size_of::<pc_price_t>();
+    match account_len {
+        price_t_size => {
+            //compute the number of lamports needed in the price account to update it
+            let rent: Rent = Default::default();
+            let lamports_needed: u64 = rent
+                .minimum_balance(new_account_len)
+                .saturating_sub(accounts[1].lamports());
+            //transfer lamports if necessary
+            if lamports_needed > 0 {
+                let transfer_instruction =
+                    transfer(accounts[0].key, accounts[1].key, lamports_needed);
```

Comment:
Instead of using account[0] and account[1]. Can we rename them at the beginning of the function?